### PR TITLE
Replaced parameter "locale" with "lang" in generated URL

### PR DESCRIPTION
**Pull request by bibich**
The `locale` parameter in generated URLs of products, categories, ... is ambiguous with the `lang` parameter. And changing the locale in URL does not change the lang of the displayed content.

As the pages are indexed with the parameter 'locale' and not 'lang' by search engines there is potentialy a problem. In fact, it only happens when the rewriting is disabled on your website, which is rather uncommon.

I 'm not sure that there won't be side effects with this PR. Will you please look it over ?

### CODE REVIEW

File: core/lib/Thelia/Core/Stack/ParamInitMiddleware.php [hunk 1]
```diff
@@ -134,7 +134,14 @@ protected function defineCurrency(TheliaRequest $request)
     protected function detectLang(TheliaRequest $request)
     {
         // first priority => lang parameter present in request (get or post)
-        if ($request->query->has("lang")) {
+        $requestedLangCodeOrLocale = $request->query->has("lang");
+
+        // add a fallback on locale parameter
+        if (null === $requestedLangCodeOrLocale) {
```
Comment (lunika) [inline core/lib/Thelia/Core/Stack/ParamInitMiddleware.php]: I think you have to compare with `false`. The `has` method will return `true` or `false`

File: core/lib/Thelia/Core/Stack/ParamInitMiddleware.php [hunk 2]
```diff
@@ -134,7 +134,14 @@ protected function defineCurrency(TheliaRequest $request)
...
+            $requestedLangCodeOrLocale = $request->query->has("locale");
+        }
+
+        if (null !== $requestedLangCodeOrLocale) {
```
Comment (lunika) [inline core/lib/Thelia/Core/Stack/ParamInitMiddleware.php]: same here

File: core/lib/Thelia/Core/Stack/ParamInitMiddleware.php (hunk 1 above)
Comment (lunika) [inline core/lib/Thelia/Core/Stack/ParamInitMiddleware.php]: Or don't use the `has` but the `get` method.

File: core/lib/Thelia/Core/Stack/ParamInitMiddleware.php [hunk 3]
```diff
@@ -134,7 +134,14 @@ protected function defineCurrency(TheliaRequest $request)
...
             // The lang parameter may contains a lang code (fr, en, ru) for Thelia < 2.2,
             // or a locale (fr_FR, en_US, etc.) for Thelia > 2.2.beta1
             $requestedLangCodeOrLocale = $request->query->get("lang");
```
Comment (lunika) [inline core/lib/Thelia/Core/Stack/ParamInitMiddleware.php]: You have to do something here, this line is wrong and depends on what you will do with the lines above.

Comment (bibich) [inline core/lib/Thelia/Core/Stack/ParamInitMiddleware.php]: Yes, you're right, this is nonsense :)

**Comment by gillesbourgeat** [comment]
:+1:

**Pull request merged**